src/test/java/de/tum/cit/aet/artemis/programming/icl/LocalVCLocalCIIntegrationTest.java

```diff
@@ -68,7 +70,18 @@
  * This class contains integration tests for the base repositories (template, solution, tests) and the different types of assignment repositories (student assignment, teaching
  * assistant assignment, instructor assignment).
  */
+
+// TestInstance.Lifecycle.PER_CLASS allows all test methods in this class to share the same instance of the test class.
+// This reduces the overhead of repeatedly creating and tearing down a new Spring application context for each test method.
+// This is especially useful when the test setup is expensive or when we want to share resources, such as database connections or mock objects, across multiple tests.
+// In this case, we want to share the same GitService and UsernamePasswordCredentialsProvider.
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
```

Comment:
I had the same idea but that's implicit default behavior, [see](https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-parallel-execution).